dsc/src/resolve.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use dsc_lib::configure::config_doc::Configuration;
+use dsc_lib::util::parse_input_to_json;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::io::Read;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+use tracing::{debug, info};
+
+use crate::util::DSC_CONFIG_ROOT;
+
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize, JsonSchema)]
+pub struct Include {
+    /// The path to the file to include.  Path is relative to the file containing the include
+    /// and not allowed to reference parent directories.  If a configuration document is used
+    /// instead of a file, then the path is relative to the current working directory.
+    #[serde(rename = "configurationFile")]
+    pub configuration_file: String,
+    #[serde(rename = "parametersFile")]
+    pub parameters_file: Option<String>,
+}
+
+/// Read the file specified in the Include input and return the content as a JSON string.
+///
+/// # Arguments
+///
+/// * `input` - The Include input as a JSON string.
+///
+/// # Returns
+///
+/// A tuple containing the contents of the parameters file as JSON and the configuration content
+/// as a JSON string.
+///
+/// # Errors
+///
+/// This function will return an error if the Include input is not valid JSON, if the file
+/// specified in the Include input cannot be read, or if the content of the file cannot be
+/// deserialized as YAML or JSON.
+pub fn get_contents(input: &str) -> Result<(Option<String>, String), String> {
+    debug!("Processing Include input");
+
+    // deserialize the Include input
+    let include = match serde_json::from_str::<Include>(input) {
+        Ok(include) => include,
+        Err(err) => {
+            return Err(format!("Error: Failed to deserialize Include input: {err}"));
+        }
+    };
+
+    let include_path = normalize_path(Path::new(&include.configuration_file))?;
+
+    // read the file specified in the Include input
+    let mut buffer: Vec<u8> = Vec::new();
+    match File::open(&include_path) {
+        Ok(mut file) => {
+            match file.read_to_end(&mut buffer) {
+                Ok(_) => (),
+                Err(err) => {
+                    return Err(format!("Error: Failed to read file '{include_path:?}': {err}"));
+                }
+            }
+        },
+        Err(err) => {
+            return Err(format!("Error: Failed to open included file '{include_path:?}': {err}"));
+        }
+    }
+    // convert the buffer to a string
+    let include_content = match String::from_utf8(buffer) {
```

Comment:
This reminds me that we need to put somewhere in the docs, that input documents for dsc.exe have to be in UTF-8.
This is important for non-English users.

Reply:
cc @michaeltlombardi